bitshuffle/bitshuffle.py

```diff
@@ -0,0 +1,445 @@
+    parser.add_argument("--editor", "-E",
+                        help="Editor to use for pasting packets. " +
+                        "If not specified, defaults in this order:\n" +
+                        "\t$VISUAL, $EDITOR, mimeopen, nano, vi, " +
+                        "emacs, micro, notepad, notepad++")
+
+    parser.add_argument("--compresstype", '-t',
+                        help="Type of compression to use. Defaults to bz2. " +
+                             "Ignored if decoding packets. " +
+                             "Currently supported: 'bz2', 'gzip'.")
+
+    args = parser.parse_args()
+
+    # Checks if no parameters were passed
+    if len(sys.argv[1:]) == 0:
+        parser.print_help()
+        sys.exit(1)
+
+    elif args.version:
+        print("Version: bitshuffle v{0}".format(version))
+        sys.exit(0)
+
+    # Encode & Decode inference
+    args = infer_mode(args)
+
+    # Set default values. Note that this ensures that args.input and
+    # args.output are open file handles (or crashes the script if not).
+    args = set_defaults(args)
+
+    assert isinstance(args.input, file_type)
+    assert isinstance(args.output, file_type)
+
+    # Main
+    if args.encode:
+        if args.compresstype not in ['bz2', 'gzip']:
+            parser.print_help()
+            sys.exit(1)
+        else:
+            packets = encode_file(args.input, args.chunksize,
+                                  args.compresslevel, args.compresstype)
+            for p in packets:
+                args.output.write(p)
+                args.output.write("\n\n")
+
+            args.output.flush()
+
+    elif args.decode:
+
+        # set to True for infile to be deleted after decoding
+        is_tmp = False
+        tmpfile = None
+        if stdin.isatty() and args.input is stdin:
+            # ask the user to paste the packets into $VISUAL
+            is_tmp = True
+            if not args.editor:
+                args.editor = find_editor()
+
+            if not check_for_file(args.editor):
+                print("Editor %s not found" % args.editor)
+                sys.exit(4)
+
+            stderr.write("editor is %s\n" % args.editor)
+
+            tmpfile = tempfile.mkstemp()[1]
+            with open(tmpfile, 'w') as tf:
+                tf.write("Paste your BitShuffle packets in this file. You " +
+                         "do not need to delete this message.\n\n")
+                tf.flush()
+            subprocess.call([args.editor, tmpfile])
+            args.input = open(tmpfile, 'r')
+
+        payload, checksum_ok = decode(args.input.read())
+        try:
+            # python 3
+            args.output.buffer.write(payload)
+        except Exception as e:
+            # python 2
+            args.output.write(payload)
+
+        if is_tmp and tmpfile:
+            os.remove(tmpfile)
+
+        if checksum_ok:
+            sys.exit(0)
+        else:
+            sys.exit(8)
+
+    args.input.close()
+    args.output.close()
```

Comment:
These are never reached.

Reply:
they are if `args.encode`

Reply:
duplicated this for `args.decode` in interests of merging soon. Would like to refactor in the future.